apps/ledger-live-mobile/src/families/ethereum/EthereumCustomFees/utils.tsx

```diff
@@ -1,26 +1,64 @@
 import BigNumber from "bignumber.js";
 import { getEnv } from "@ledgerhq/live-env";
 import { inferDynamicRange, Range } from "@ledgerhq/live-common/range";
-import { Transaction } from "@ledgerhq/live-common/families/ethereum/types";
+import { formatCurrencyUnit } from "@ledgerhq/coin-framework/currencies/index";
+import {
+  Transaction,
+  TransactionRaw,
+} from "@ledgerhq/live-common/families/ethereum/types";
+import { Account } from "@ledgerhq/types-live";
 
 const TWENTY_GWEI = new BigNumber(10e9);
 
 const defaultMaxPriorityFeeRange = inferDynamicRange(TWENTY_GWEI); // 0 - 20 Gwei
 const defaultMaxFeePerGasRange = inferDynamicRange(TWENTY_GWEI); // 0 - 20 Gwei
 
 export const inferMaxPriorityFeeRange = (
+  mainAccount: Account,
   networkInfo: Transaction["networkInfo"],
+  transactionRaw?: TransactionRaw,
 ): Range => {
   if (!networkInfo?.maxPriorityFeePerGas || !networkInfo?.nextBaseFeePerGas)
     return defaultMaxPriorityFeeRange;
 
+  let minValue = getEnv("EIP1559_MINIMUM_FEES_GATE")
+    ? networkInfo.maxPriorityFeePerGas.min.times(
+        getEnv("EIP1559_PRIORITY_FEE_LOWER_GATE"),
+      )
+    : new BigNumber(0);
+
+  let maxValue = networkInfo.maxPriorityFeePerGas.max;
+
+  if (transactionRaw?.maxPriorityFeePerGas) {
+    const maxPriorityFeeGap: number = getEnv(
+      "EDIT_TX_EIP1559_MAXPRIORITYFEE_GAP_SPEEDUP_FACTOR",
+    );
+    const newMaxPriorityFeePerGas = new BigNumber(
+      transactionRaw.maxPriorityFeePerGas,
+    ).times(1 + maxPriorityFeeGap);
+
+    const { units } = mainAccount.currency;
+    const unit = units.length > 1 ? units[1] : units[0];
+    const newMaxPriorityFeePerGasinGwei = new BigNumber(
```

Comment:
Not sure about the name either `*inGwei`, you can't know what unit is going to be used here as `unit` can be anything. 
For `ethereum` yes `unit` will be Gwei, but for polygon, it's still gonna be a `1e18` unit (so still MATIC here)

Reply:
Not sure the understand if the issue is the naming of the variable or the usage of `units` variable